packages/common/src/audius-query/createApi.ts

```diff
@@ -145,45 +146,59 @@ const buildEndpointHooks = <
       // Retrieve data from cache if lookup args provided
       if (!endpointState[key]) {
         if (
-          !(endpoint.options?.idArgKey || endpoint.options?.permalinkArgKey) ||
+          !(
+            endpoint.options?.idArgKey ||
+            endpoint.options?.idListArgKey ||
+            endpoint.options?.permalinkArgKey
+          ) ||
           !endpoint.options?.kind ||
           !endpoint.options?.schemaKey
         )
           return null
-        const { kind, idArgKey, permalinkArgKey, schemaKey } = endpoint.options
-        if (idArgKey && !fetchArgs[idArgKey]) return null
-        if (permalinkArgKey && !fetchArgs[permalinkArgKey]) return null
+        const { kind, idArgKey, idListArgKey, permalinkArgKey, schemaKey } =
+          endpoint.options
 
-        const idAsNumber = idArgKey
-          ? typeof fetchArgs[idArgKey] === 'number'
-            ? parseInt(fetchArgs[idArgKey])
-            : fetchArgs[idArgKey]
-          : null
-        const idCachedEntity = idAsNumber
-          ? cacheSelectors.getEntry(state, {
-              kind,
-              id: idAsNumber
-            })
-          : null
-
-        let permalinkCachedEntity = null
-        if (kind === Kind.TRACKS && permalinkArgKey) {
-          permalinkCachedEntity = getTrack(state, {
-            permalink: fetchArgs[permalinkArgKey]
-          })
-        }
-        if (kind === Kind.COLLECTIONS && permalinkArgKey) {
-          permalinkCachedEntity = getCollection(state, {
-            permalink: fetchArgs[permalinkArgKey]
```

Comment:
@sddioulde Just a callout that I refactored some of the `permalinkArgKey` logic. Pretty sure it's still working, but if you run into trouble after merging, lmk